### ISSUE TITLE
Manual backend initialization crashes if no windows present

### ISSUE THREAD

**Issue by heyer2** [glumpy/glumpy#81]
Title says it all.
The following code works:

`from glumpy import app, gloo, gl

win = app.Window()
backend = app.__backend__
clock = app.__init__(backend=backend)`

The following doesn't:

`from glumpy import app, gloo, gl

#win = app.Window()
backend = app.__backend__
clock = app.__init__(backend=backend)`

Using OSX with GLFW backend

**Comment by rougier** [comment]
Interesting, the line `backend = app.backend` does not work for me, there is no backend.
Can you print the backend ?

**Comment by heyer2** [comment]
My apologies, somehow Github removed the underscores, here it is:

backend = app.__backend__
clock = app.__init__(backend=backend)

**Comment by rougier** [comment]
I'm not sure I get the wrong script (from you comment they appear to be the same). You can use triple backquote to surround your code by the way. Can you write again the faulty script ?

This one below works for me:

```
from glumpy import app, gloo, gl
win = app.Window()
backend = app.__backend__
clock = app.__init__(backend=backend)
```

**Comment by heyer2** [comment]
Yes, you got it right.
I see you manually specifying the backend is a necessity.
How come opening a window enables a backend?
Perhaps there should be a warning indicating a lack of backend, it sure confused me.

**Comment by heyer2** [comment]
I just tested to be sure. This code that worked for you does not work for me:
`from glumpy import app, gloo, gl
app.use("glfw")
backend = app.__backend__
clock = app.__init__(backend=backend)`

**Comment by rougier** [comment]
Yes, you're right, sorry, I got a segfault as well. I think the backend needs to be initialized which is made during window creation but not with the clock. I'll look into that.

**Comment by rougier** [comment]
By the way, why do you want to initialize the app without a window ?

**Comment by heyer2** [comment]
For my application I dynamically open none to several windows.
Initializing the backend together with all the other initialization code just seemed convenient in comparison to manually checking if a window was opened.

**Comment by rougier** [comment]
Problem is that in the ```__init__``` function, there is some GL initialization code that will fail (segfault) if no window has been created because it means no GL context exist. Can you check if commenting lines starting with ```gl.```fixes the problem ?

**Comment by heyer2** [comment]
I checked that earlier actually, it did.

**Comment by rougier** [comment]
Ok, then maybe the GL initialization can be during window creation. The only problem is to ensure this is done only once / GL context.
